avocado/utils/ssh.py

```diff
@@ -168,6 +173,14 @@ def connect(self):
             self._connection = master
         return self._check()
 
+    @property
+    def control_master(self):
+        control = "~/.ssh/avocado-master-{}@{}:{}".format(self.user,
+                                                          self.host,
+                                                          self.port)
+        if os.path.exists(control):
```

Comment:
@beraldoleal this approach is fine, but code is not working properly. as control is "~/.ssh/avocado-master-root@121.1.1.88:22"  so  both os.path.exists() and os.unlink() will fail. so its can be like this,
```
    def control_master(self):
        control = "~/.ssh/avocado-master-{}@{}:{}".format(self.user,
                                                          self.host,
                                                          self.port)
        control_path = os.path.expanduser(control)
        if os.path.exists(control_path):
            return control_path
```

Reply:
You are right @smitterl , thanks for this catch. I will fix on a force push.

Reply:
now its fine.